packages/graphql-language-service-interface/src/getAutocompleteSuggestions.ts

```diff
@@ -246,6 +248,16 @@ export function getAutocompleteSuggestions(
     return getSuggestionsForVariableDefinition(token, schema, kind);
   }
 
+  // Input value definition
+  if (kind === RuleKinds.INPUT_VALUE_DEF && step === 2) {
```

Comment:
I'm not particularly sure what step means here, but I just found it to be representative of the current state. Hope to get some advice :)

Reply:
yep, that's it! essentially steps for a parser rule to advance.

here's where the magic happens in parser

https://github.com/graphql/graphiql/blob/c4cba85e869105f5363ebcb3826b4cf7b7e62c00/packages/graphql-language-service-parser/src/onlineParser.ts#L256-L256

this is probably the most straightforward implementation example. with ARGUMENT, step 1 is when we want argument name, and step 2 is when we want an input value. 

https://github.com/graphql/graphiql/blob/c4cba85e869105f5363ebcb3826b4cf7b7e62c00/packages/graphql-language-service-interface/src/getAutocompleteSuggestions.ts#L161

so, if i wanted to implement autocompletion for field names for object types that extended interfaces, i would do that on `OBJECT_TYPE` step 1.

Reply:
oh and i guess you found one that has three steps! :D

Reply:
Just thinking aloud.. is it possible for the steps to have constants for referencing them? In order to give the steps meaning beyond just `2` for example? 🤔

Reply:
good question! they have a different meaning in the context of each rule. we could have `Step.First` but that wouldn't be much more readable, or would it?

Reply:
Not really, it would pretty much be the same thing 😄 I'll think about it a bit more.

Reply:
we could also specify enums for each rule that has multiple steps that are referenced, such as `ObjectTypeSteps.Name = 1` and `ObjectTypeSteps.Field = 2`, and then `FragmentDefintionSteps.Name = 1`, etc

Reply:
Maybe that would make sense, since the meaning of each step is defined by the GraphQL type being worked on. My only concern is that, since the values overlap, there might be an issue where they are mixed up. Typescript wouldn't complain since they are all numbers (except they are typed with the right enum type 🤔 ) but the behavior wouldn't be the expected behavior.

For example, `InputObjectTypeSteps.Field` might be equal to `RuleKindStep.Name`. Maybe my concern is a non-issue though. Perhaps if there is a way to type guard `step` to one of the enums, that would make `step` have just one meaning and remove ambiguity.